core/src/main/java/com/linecorp/armeria/server/internal/annotation/AnnotatedHttpDocServicePlugin.java

```diff
@@ -0,0 +1,322 @@
+    public Set<Class<? extends Service<?, ?>>> supportedServiceTypes() {
+        return ImmutableSet.of(AnnotatedHttpService.class);
+    }
+
+    @Override
+    public ServiceSpecification generateSpecification(Set<ServiceConfig> serviceConfigs) {
+        requireNonNull(serviceConfigs, "serviceConfigs");
+        final Map<Class<?>, Set<MethodInfo>> methodInfos = new HashMap<>();
+        serviceConfigs.forEach(sc -> {
+            final Optional<AnnotatedHttpService> service = sc.service().as(AnnotatedHttpService.class);
+            service.ifPresent(
+                    httpService -> addMethodInfo(methodInfos, sc.virtualHost().hostnamePattern(), httpService));
+        });
+
+        return generate(methodInfos);
+    }
+
+    private static void addMethodInfo(Map<Class<?>, Set<MethodInfo>> methodInfos,
+                                      String hostnamePattern, AnnotatedHttpService service) {
+        final PathMapping pathMapping = service.pathMapping();
+        final EndpointInfo endpoint = endpointInfo(pathMapping, hostnamePattern);
+        if (endpoint == null) {
+            return;
+        }
+
+        final Method method = service.method();
+        final String name = method.getName();
+        final TypeSignature returnTypeSignature = toTypeSignature(method.getGenericReturnType());
+        final List<FieldInfo> fieldInfos = fieldInfos(service.annotatedValueResolvers());
+        final Class<?> clazz = service.object().getClass();
+        pathMapping.supportedMethods().forEach(
+                httpMethod -> {
+                    final MethodInfo methodInfo = new MethodInfo(
+                            name, returnTypeSignature, fieldInfos, ImmutableList.of(), // Ignore exceptions.
+                            ImmutableList.of(endpoint), httpMethod);
+                    methodInfos.computeIfAbsent(clazz, unused -> new HashSet<>()).add(methodInfo);
+                });
+    }
+
+    @Nullable
+    @VisibleForTesting
+    static EndpointInfo endpointInfo(PathMapping pathMapping, String hostnamePattern) {
+        final String endpointPath;
+        String regexPathPrefix = null;
+        if (pathMapping.prefix().isPresent()) {
+            if (pathMapping.regex().isPresent()) { // PrefixAddingPathMapping.
+                regexPathPrefix = pathMapping.prefix().get();
+                endpointPath = REGEX + pathMapping.regex().get();
+            } else { // PrefixPathMapping.
+                endpointPath = PREFIX + pathMapping.prefix().get();
+            }
+        } else {
+            endpointPath = getNormalizedTriePathOrRegex(pathMapping);
+        }
+
+        if (isNullOrEmpty(endpointPath)) {
+            return null;
+        }
+
+        final EndpointInfoBuilder builder = new EndpointInfoBuilder(hostnamePattern, endpointPath);
+        if (!isNullOrEmpty(regexPathPrefix)) {
+            builder.regexPathPrefix(regexPathPrefix);
+        }
+
+        builder.defaultMimeType(MediaType.JSON_UTF_8);
```

Comment:
If I understand correctly, doesn't it make sense to set `MediaType.ANY_TYPE` as a default when `consumeTypes.isEmpty() == true`?